src/form/input/Input.tsx

```diff
@@ -58,195 +57,201 @@ export type InputProps = Omit<
   };
 };
 
-/* eslint-disable complexity */
-function Input(props: InputProps) {
-  const {
-    testid,
-    value,
-    type = "text",
-    isDisabled,
-    hasError,
-    customClassName,
-    leftIcon,
-    rightIcon,
-    localizationOptions = {},
-    role,
-    autoComplete = "off",
-    autoCorrect = "off",
-    inputContainerRef,
-    onChange,
-    ...rest
-  } = props;
-  const {
-    shouldFormatToLocaleString = false,
-    locale,
-    maximumFractionDigits = 0
-  } = localizationOptions;
-  const [
-    {
-      DECIMAL_NUMBER_SEPARATOR: decimalSeparatorForLocale,
-      MINUS_SIGN: minusSignForLocale,
-      LOCALE_NEGATIVE_ZERO: negativeZeroForLocale
-    },
-    setNumberSeparatorsForLocale
-  ] = useState(() => ({
-    ...getNumberSeparators(locale),
-    ...getNegativeZero(locale)
-  }));
-  const isNumberInput = type === "number";
-  const inputContainerClassName = classNames(
-    "input-container",
-    customClassName,
-    `input-container--type-${type}`
-  );
-  const inputClassName = classNames("input", {
-    "input--is-disabled": isDisabled,
-    "input--has-error": hasError
-  });
-  let finalValue = value;
-
-  if (
-    !(
-      typeof maximumFractionDigits === "number" &&
-      Number.isInteger(maximumFractionDigits) &&
-      maximumFractionDigits >= 0
-    )
-  ) {
-    throw new Error("maximumFractionDigits should be zero or a positive integer.");
-  }
-
-  if (isNumberInput && typeof value === "string" && shouldFormatToLocaleString) {
-    const [integerPart, decimalPart] = String(value).split(".");
-    const numberFormatter = formatNumber({
-      maximumFractionDigits,
-      locale
+const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  /* eslint-disable complexity */
+  (props: InputProps, ref) => {
```

Comment:
I think TSC can infer the type for props from by the definition of `React.forwardRef<HTMLInputElement, InputProps>`
```suggestion
  (props, ref) => {
```